diff --git a/qutebrowser/config/configtypes.py b/qutebrowser/config/configtypes.py
index 7b5125b1c..98823e278 100644
--- a/qutebrowser/config/configtypes.py
+++ b/qutebrowser/config/configtypes.py
@@ -1001,16 +1001,17 @@ class QtColor(BaseType):
     * `hsv(h, s, v)` / `hsva(h, s, v, a)` (values 0-255, hue 0-359)
     """
 
-    def _parse_value(self, val: str) -> int:
+    def _parse_value(self, val: str, maxval: int = 255) -> int:
         try:
             return int(val)
         except ValueError:
             pass
 
-        mult = 255.0
         if val.endswith('%'):
             val = val[:-1]
-            mult = 255.0 / 100
+            mult = maxval / 100
+        else:
+            mult = maxval / 1.0
 
         try:
             return int(float(val) * mult)
@@ -1029,14 +1030,19 @@ class QtColor(BaseType):
             openparen = value.index('(')
             kind = value[:openparen]
             vals = value[openparen+1:-1].split(',')
-            int_vals = [self._parse_value(v) for v in vals]
-            if kind == 'rgba' and len(int_vals) == 4:
+            if kind == 'rgba' and len(vals) == 4:
+                int_vals = [self._parse_value(v) for v in vals]
                 return QColor.fromRgb(*int_vals)
-            elif kind == 'rgb' and len(int_vals) == 3:
+            elif kind == 'rgb' and len(vals) == 3:
+                int_vals = [self._parse_value(v) for v in vals]
                 return QColor.fromRgb(*int_vals)
-            elif kind == 'hsva' and len(int_vals) == 4:
+            elif kind == 'hsva' and len(vals) == 4:
+                int_vals = ([self._parse_value(vals[0], maxval=359)] +
+                            [self._parse_value(v) for v in vals[1:]])
                 return QColor.fromHsv(*int_vals)
-            elif kind == 'hsv' and len(int_vals) == 3:
+            elif kind == 'hsv' and len(vals) == 3:
+                int_vals = ([self._parse_value(vals[0], maxval=359)] +
+                            [self._parse_value(v) for v in vals[1:]])
                 return QColor.fromHsv(*int_vals)
             else:
                 raise configexc.ValidationError(value, "must be a valid color")
