diff --git a/qutebrowser/browser/webkit/network/networkreply.py b/qutebrowser/browser/webkit/network/networkreply.py
index 6f7abeaab..4fb7dfea5 100644
--- a/qutebrowser/browser/webkit/network/networkreply.py
+++ b/qutebrowser/browser/webkit/network/networkreply.py
@@ -116,7 +116,7 @@ class ErrorNetworkReply(QNetworkReply):
         # the device to avoid getting a warning.
         self.setOpenMode(QIODevice.OpenModeFlag.ReadOnly)
         self.setError(error, errorstring)
-        QTimer.singleShot(0, lambda: self.error.emit(error))
+        QTimer.singleShot(0, lambda: self.errorOccurred.emit(error))
         QTimer.singleShot(0, lambda: self.finished.emit())
 
     def abort(self):
diff --git a/tests/unit/browser/webkit/network/test_networkreply.py b/tests/unit/browser/webkit/network/test_networkreply.py
index 5de31bf0f..fc400c1b2 100644
--- a/tests/unit/browser/webkit/network/test_networkreply.py
+++ b/tests/unit/browser/webkit/network/test_networkreply.py
@@ -78,7 +78,7 @@ def test_error_network_reply(qtbot, req):
     reply = networkreply.ErrorNetworkReply(
         req, "This is an error", QNetworkReply.NetworkError.UnknownNetworkError)
 
-    with qtbot.wait_signals([reply.error, reply.finished], order='strict'):
+    with qtbot.wait_signals([reply.errorOccurred, reply.finished], order='strict'):
         pass
 
     reply.abort()  # shouldn't do anything
